### ISSUE TITLE
Post Ranking Algorithm

### ISSUE THREAD

**Issue by ghost** [platf0rm/platform#16]
Every posts will initially have 1 vote owned by the original author of the post (comment or thread).

```
Rank = (v- 1) / t
```
Where,
`v` = total votes in a post
`t` = age of the post in hours

In action ([interfactive](http://www.wolframalpha.com/input/?i=plot%28+%2820+-+1%29+%2F+t+,+%2830+-+1%29+%2F+t+,+%2860+-+1%29+%2F+t+%29+where+t+%3D+0..24)):
![algo_graph](https://cloud.githubusercontent.com/assets/15170164/24347205/5e78abe6-128c-11e7-8119-2bde2353c0ba.gif)

We can also put weights to slow-down or speed-up the decay of (older) post's ranking score.

```
Rank = (v - 1) / t ^ w
```
Where `w` = the weight or gravity, optimal weight is `1.6` or lower.

Any thoughts?

**Comment by TheoryOfNekomata** [comment]
Is the post ranking affected by the number of comments?

**Comment by ghost** [comment]
No, not as of the moment. We first need to know the variables in the post's comment activity that would affect it's ranking score.

I like Reddit's approach in ranking and sorting the comments and how it affects the thread's overall ranking score: [clickable](https://redditblog.com/2009/10/15/reddits-new-comment-sorting-system/)

**Comment by makoru-hikage** [comment]
We'll base it first on votes. To put comments in the equation, we must mind also the votes in a given comment.

**Comment by ghost** [comment]
This algorithm is not specific to threads. Threads and comments are posts themselves and thus this algorithm will also apply to a thread's comment with the exception of "the age of the post" variable so the comments will be sorted by votes rather than votes / age.
